### ISSUE TITLE
vfolder windows should be per-mailbox, resetting to present when changing mailbox

### ISSUE THREAD

**Issue by ebblake** [neomutt/neomutt#3014]
## What would you like to be able to do?
When using <vfolder-window-backward> and friends, right now, the choice of date window is global and persists even when changing to a different mailbox.  I think it would be more intuitive if the window was remembered per-mailbox, where changing to a different mailbox opens up in the present regardless of what window had been active in the previous mailbox.

## What needs to change in NeoMutt?
finish the work towards having per-mailbox and per-account configs that can be separate from global configs (yes, that's been a big ongoing task), then tweak the vfolder windowing configs to be per-mailbox instead of global.

## Have you tried any alternative methods?
This is somewhat of a followup to issue #2972.  Now that we have a function for resetting the window to present, it would be easy enough to bind `c` to a macro that resets to present before changing mailbox, but having that reset be automatic without having to add the binding might make the use of windowing simpler out of the box.

**Comment by flatcap** [comment]
> changing to a different mailbox opens up in the present
> regardless of what window had been active in the previous mailbox.

Hmm... I'd have imagined that an existing window would remain the same.
If you switch back and forth, nothing changes.

Either way, a function to describe the current window might be useful:

`folder-hook notmuch:// "push <vfolder-window-describe>"`

> **Window size: 1 week, viewing 3 months ago, 15 emails**

**Comment by ebblake** [comment]
> Hmm... I'd have imagined that an existing window would remain the same.
> If you switch back and forth, nothing changes.

Yeah, for revisiting an existing folder, preserving the window makes sense; it is opening a new folder where starting from the present is most important.  So if I set the 'tag:inbox' mailbox to last month, then jump to a folder for tag:neomutt (in the present), then jump back to tag:inbox, I would return to last month's window.

> 
> Either way, a function to describe the current window might be useful:
> 
> `folder-hook notmuch:// "push <vfolder-window-describe>"`
> 
> > **Window size: 1 week, viewing 3 months ago, 15 emails**

Or at a minimum, an expando in `$status_format`, or even a way to do date-sensitive coloration of the status line, so that it becomes visually apparent whether the folder is in the present or in a prior window.

**Comment by jindraj** [comment]
>Or at a minimum, an expando in $status_format, or even a way to do date-sensitive coloration of the status line, so that it becomes visually apparent whether the folder is in the present or in a prior window.

I do have something similar. See my config file and asciinema cast below
https://github.com/jindraj/dotfiles/blob/df69fa80afc8aa51e6c2c5025cb19c345c8cbf8b/.neomutt/neomuttrc.profile-common#L5-L18

[![asciicast](https://asciinema.org/a/hv3mQHHr6lh33T9EpZSHjRZ8t.svg)](https://asciinema.org/a/hv3mQHHr6lh33T9EpZSHjRZ8t)
